### PR TITLE
#692 And/Or javadoc

### PR DESCRIPTION
**Pull request by Vatavuk**
Expanded javadoc for And/Or/AndInThreads for #692 

I've updated `And` and `Or` javadoc and added a link to `AndInThreads` which refers to `And` javadoc examples.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#817](https://github.com/yegor256/cactoos/issues/817) is assigned to @krzyk[/z](https://www.0crat.com/u/krzyk), here is [why](https://www.0crat.com/footprint/C63314D6Z/1241859650); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer

File: src/main/java/org/cactoos/scalar/Or.java [hunk 1]
```diff
@@ -34,6 +34,36 @@
...
  * Logical disjunction.
  *
+ * <p>This class can be effectively used to iterate through
+ * a collection, just like
+ * {@link java.util.stream.Stream#forEach(java.util.function.Consumer)}
+ * works:</p>
+ *
+ * <pre>
+ * new Or(
+ *    new ProcOf<>(input -> System.out.printf("\'%s\' ", input) ),
```
Comment (krzyk) [inline src/main/java/org/cactoos/scalar/Or.java]: @Vatavuk You can't use `<` or `>` inside javadoc without some kind of escaping, either `&lt;` and `&gt;` or preferably replace the `<pre>...</pre>` with `{@code ...}`

**Comment by Vatavuk** [comment]
@krzyk fixed in abf8ec8

File: src/main/java/org/cactoos/scalar/Or.java [hunk 2]
```diff
@@ -34,6 +34,40 @@
 /**
  * Logical disjunction.
  *
+ * <p>This class can be effectively used to iterate through
+ * a collection, just like
+ * {@link java.util.stream.Stream#forEach(java.util.function.Consumer)}
+ * works:</p>
+ *
+ * <pre>
```
Comment (krzyk) [inline src/main/java/org/cactoos/scalar/Or.java]: @Vatavuk pre now should be removed (`{@code }` already uses code font)

File: src/main/java/org/cactoos/scalar/Or.java [hunk 3]
```diff
@@ -34,6 +34,40 @@
...
+ *    new IterableOf<>("Mary", "John", "William", "Napkin")
+ * ).value(); // will print 'Mary' 'John' 'William' 'Napkin' to standard output
+ *            // the result of this operation is always false
+ * }
+ * </pre>
+ *
+ * <p>This class could be also used for matching multiple boolean
+ * expressions:</p>
+ *
+ * <pre>
```
Comment (krzyk) [inline src/main/java/org/cactoos/scalar/Or.java]: @Vatavuk and here also, `<pre></pre>` should be removed

**Comment by Vatavuk** [comment]
@krzyk fixed in ee45ae1

**Comment by krzyk** [comment]
@llorllale ready to merge

**Comment by Vatavuk** [comment]
@llorllale ping

File: src/main/java/org/cactoos/scalar/And.java [hunk 4]
```diff
@@ -38,12 +38,30 @@
...
  * new And(
- *    new ProcOf<>(input -> System.out.printf("\'%s\' ", input) ),
+ *    new ProcOf<>(input -> System.out.printf("\'%s\' ", input)),
  *    new IterableOf<>("Mary", "John", "William", "Napkin")
  * ).value(); // will print 'Mary' 'John' 'William' 'Napkin' to standard output
- * </pre>
+ *            // the result of this operation is always true
+ * }
+ *
+ * <p>This class could be also used for matching multiple boolean
```
Comment (llorllale) [inline src/main/java/org/cactoos/scalar/And.java]: @Vatavuk could you also please add a note somewhere that And performs a short-circuit evaluation?

File: src/main/java/org/cactoos/scalar/Or.java [hunk 5]
```diff
@@ -34,6 +34,36 @@
...
+ *
+ * {@code
+ * new Or(
+ *    new ProcOf<>(input -> System.out.printf("\'%s\' ", input) ),
+ *    new IterableOf<>("Mary", "John", "William", "Napkin")
+ * ).value(); // will print 'Mary' 'John' 'William' 'Napkin' to standard output
+ *            // the result of this operation is always false
+ * }
+ *
+ * <p>This class could be also used for matching multiple boolean
```
Comment (llorllale) [inline src/main/java/org/cactoos/scalar/Or.java]: @Vatavuk could you also please add a note somewhere that Or performs a short-circuit evaluation?

**Comment by llorllale** [comment]
@Vatavuk two minor observations and we're good, see above

**Comment by Vatavuk** [comment]
@llorllale fixed in f774130

**Comment by llorllale** [comment]
@rultor merge

**Comment by Vatavuk** [comment]
@llorllale Can you please merge again, it seems that we have also a problem with stability of `AndInThreadsTest `. These tests are failing randomly on my local machine also.

**Comment by llorllale** [comment]
@Vatavuk you're right. Please open a bug

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
@elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova) please review this job completed by @krzyk[/z](https://www.0crat.com/u/krzyk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#817](https://github.com/yegor256/cactoos/issues/817) is now out of scope

**Comment by elenavolokhova** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @krzyk[/z](https://www.0crat.com/u/krzyk)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova)

**Pull request merged**